src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -343,6 +356,53 @@ private void validateOperationProperties(@Nullable Schema schema, FailureCollect
           .withConfigElement(NAME_TABLE_KEY, keyField);
       }
     }
+
+    Map<String, Integer> keyMap = calculateDuplicates(keyFields);
+    keyMap.keySet().stream()
+      .filter(key -> keyMap.get(key) != 1)
+      .forEach(key -> collector.addFailure(
+        String.format("Table key field '%s' is duplicated.", key),
+        String.format("Remove duplicates of Table key field '%s'.", key))
+        .withConfigElement(NAME_TABLE_KEY, key)
+      );
+
+    if (Operation.UPSERT.equals(getOperation()) && getOrderedBy() != null) {
+      List<String> orderedByList = Arrays.stream(Objects.requireNonNull(getOrderedBy()).split(","))
+        .map(String::trim).collect(Collectors.toList());
+      LOG.debug("-> " + orderedByList);
+      orderedByList.stream()
+        .filter(v -> !fields.contains(v.split(":")[0]))
```

Comment:
can we use a space instead of ':'? That way it translates directly to what would be used in the query.